src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobExecutionService.java

```diff
@@ -453,21 +456,45 @@ private BuildResult constructBuildResult(List<BuildResult.LocalCITestJobDTO> fai
     }
 
     private Path cloneRepository(VcsRepositoryUri repositoryUri, @Nullable String commitHash, boolean checkout, String buildJobId) {
+        int attempt = 0;
+        Repository repository = null;
+
+        // Ignore warning about always true condition, as the loop will break if the clone is successful
```

Comment:
What warning? This loop is guaranteed to terminate, as the first thing that happens is that we increment `attempt`.

Reply:
Intellij was showing a warning that this condition is always true, deleted the comment since it's displayed for you